src/libraries/System.Net.Http/src/System/Net/Http/HttpRequestException.cs

```diff
@@ -39,6 +35,27 @@ public HttpRequestException(string? message, Exception? inner, HttpStatusCode? s
             StatusCode = statusCode;
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HttpRequestException" /> class with a specific message an inner exception, and an HTTP status code and an <see cref="HttpRequestError"/>.
+        /// </summary>
+        /// <param name="message">A message that describes the current exception.</param>
+        /// <param name="inner">The inner exception.</param>
+        /// <param name="statusCode">The HTTP status code.</param>
+        /// <param name="httpRequestError">The <see cref="HttpRequestError"/> that caused the exception.</param>
+        public HttpRequestException(string? message, Exception? inner = null, HttpStatusCode? statusCode = null, HttpRequestError? httpRequestError = null)
+            : this(message, inner, statusCode)
+        {
+            HttpRequestError = httpRequestError;
+        }
+
+        /// <summary>
+        /// Gets the <see cref="Http.HttpRequestError"/> that caused the exception.
+        /// </summary>
+        /// <value>
+        /// The <see cref="Http.HttpRequestError"/> or <see langword="null"/> if the underlying <see cref="HttpMessageHandler"/> did not provide it.
+        /// </value>
+        public HttpRequestError? HttpRequestError { get; }
```

Comment:
Why is this nullable if we have Unknown, or the other way around?

Reply:
> and then agreeing with the advice from the API review group that nullable actually makes sense.

I don't understand why nullable makes sense here. I was away for the API review, and see the comments in the issue:

> Changed the HttpRequestError property on HttpRequestException to be nullable, to distinguish "the provider didn't give one" from "the provider had an error it couldn't map".

but I agree that this is a confusing distinction to try to make.  What action is someone supposed to take differently based on null vs Unknown?

cc: @bartonjs, @terrajobst

Reply:
The main distinction that I can think of would be "I should try again if the error is [some specific code(s)], and not if not, and Unknown is clearly not one of those codes".  If it's null (vs Unknown) then the caller can decide "the underlying provider doesn't know how to set this code" is different from "something went wrong that the caller didn't know how to map to this enum"

Reply:
I still don't understand the distinction / how someone would apply it.  You're saying someone would write code that would retry on null but not on Unknown, or retry on Unknown but not null?  Whether the provider knows what the problem is but can't map it, or doesn't know and can't map it, the result is the same: the error is unknown and you as the consumer have zero additional information to inform you as to what to do with it.  I'm not understanding why we'd choose two different representations for that same case.